packages/@aws-cdk/aws-s3-assets/lib/asset.ts

```diff
@@ -103,6 +164,40 @@ export class Asset extends cdk.Construct implements assets.IAsset {
   constructor(scope: cdk.Construct, id: string, props: AssetProps) {
     super(scope, id);
 
+    // Bundling
+    let bundleAssetPath: string | undefined;
+    if (props.bundling) {
+      // Create the directory for the bundle next to the asset path
+      const bundlePath = path.join(path.dirname(props.path), props.bundling.bundleDirectory ?? '.bundle');
```

Comment:
1. I think the default bundle directory should be `.$basename.bundle` in case there are multiple sources.
2. If `props.bundleDirectory` is provided, it should be respected as-is (and not as a directory next to `props.path`. For example, if a user wants to put this in a temp directory, they should be able to to pass `/tmp/my-bundle` and it should just work.

Something like this:

```suggestion
      const bundlePath = props.bundling.bundleDirectory ?? path.join(path.dirname(props.path),  `.${path.basename(props.path)}.bundle`);
```

Reply:
I wonder if can should just use a fresh /tmp directory instead for the default?

Reply:
`bundlePath` is the main dir for bundles, in it I create `bundleAssetPath`.

For the python integ test in Lambda, we have the source at `test/python-lambda-handler` and the output is at `test/.bundle/python-lambda-handler`.

You saying it should be at `test/.python-lambda-handler.bundle`? or somewhere in `/tmp`? As a user I find it nice to have it close to the source so I can have a look at the content of it. Having it in `.bundle` with subfolders inside it is easier to gitignore?

Reply:
Maybe `.bundle/<node unique id>` because the same asset path can be used for multiple assets?

Reply:
yes, let's do `.bundle/<unique-id>` and also default to simply `.bundle` (i.e. relative to CWD and not to the asset sources). Consider that asset sources can also come from some 3rd party dependency that are mounted in a read-only file-system (yarn2).